### PR TITLE
make key algorithm configurable for local development with in memory …

### PR DESCRIPTION
**Pull request by endophage**
…ed25519 crypto service

Also bugsnag 

@dmcgowan 

Signed-off-by: David Lawrence <david.lawrence@docker.com> (github: endophage)

### CODE REVIEW

File: cmd/notary-server/dev-config.json [hunk 1]
```diff
@@ -7,7 +7,8 @@
 	"trust_service": {
 		"type": "local",
 		"hostname": "",
-		"port": ""
+		"port": "",
+		"key_algorithm": "ed25519"
```
Comment (diogomonica) [inline cmd/notary-server/dev-config.json]: why ed?

**Comment by dmcgowan** [comment]
So I take it now setting the algorithm to `"ed25519"` still works as expected since it is not attempting to hardcode to ecdsa?

LGTM

File: cmd/notary-server/dev-config.json (hunk 1 above)
Comment (endophage) [inline cmd/notary-server/dev-config.json]: Only algo supported for local trust service

Comment (endophage) [inline cmd/notary-server/dev-config.json]: Note that's in dev-config.json which is set up to run a notary server locally, fully in memory, outside a container.

**Comment by endophage** [comment]
@dmcgowan after this latest commit that should be the case.

**Comment by endophage** [comment]
@NathanMcCauley @diogomonica @aaronlehmann can one of you LGTM or otherwise provide feedback on this so we can merge it.

File: errors/errors.go [hunk 2]
```diff
@@ -10,70 +10,69 @@ import (
 // with the registry API
 const errGroup = "notary.api.v1"
 
+// These errors should be returned from contextHandlers only. They are
+// serialized and returned to a user as part of the generic error handling
+// done by the rootHandler
 var (
-	// ErrNoStorage lint comment
```
Comment (endophage) [inline errors/errors.go]: @dmcgowan worked out how to get rid of all these comments. Apparently lint accepts a single comment before the var/const block :-)

**Comment by diogomonica** [comment]
LGTM

**Pull request merged**